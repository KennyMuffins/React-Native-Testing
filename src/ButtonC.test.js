import React from 'react';
import {shallow} from 'enzyme';
import ButtonC from './ButtonC';

let mockOpenURL = jest.fn();

// 1. Set openURL module function to jest.fn
jest.mock('react-native/Libraries/Linking/Linking', () => ({
    openURL: mockOpenURL,
}));

describe('ButtonC', () => {
    describe('Rendering', () => {
        it('should match to snapshot - Primary', () => {
            const component = shallow(<ButtonC label="test label" primary />)
            expect(component).toMatchSnapshot("Primary button snapshot")
        });
        it('should match to snapshot - Secondary', () => {
            const component = shallow(<ButtonC label="test label" primary={false} />)
            expect(component).toMatchSnapshot("Secondary button snapshot")
        });
    });

    describe('Interaction', () => {
        describe('onPressHandler', () => {
            const mockOnPress = jest.fn();
            let instance;

            // 2. preparing a reusable shallow rendered instance
            beforeEach(() => {
                instance = shallow(<ButtonC
                    label="test label" 
                    onPress={mockOnPress}
                    url="https://www.test.com"
                />).instance();
                jest.clearAllMocks();
            });

            it('should call onPress', () => {
                instance.onPressHandler();
                expect(mockOnPress).toHaveBeenCalled();
                expect(mockOnPress).toHaveBeenCalledTimes(1);
            });

            it('should call openURL if url is provided', () => {
                instance.onPressHandler();
                // 3. mockOpenURL should be called since we have passed in 'url' prop
                expect(mockOpenURL).toHaveBeenCalled();
                expect(mockOpenURL).toHaveBeenCalledTimes(1);
                expect(mockOpenURL).toHaveBeenCalledWith("https://www.test.com");
            });

            it('should not call openURL if url is nor provided', () => {
                const innerInstance = shallow(<ButtonC label="test label" onPress={mockOnPress} />).instance();
                innerInstance.onPressHandler();
                // 4. mockOpenURL should NOT be called as we have NOT passed in 'url' prop
                expect(mockOpenURL).not.toHaveBeenCalled();
            });
        });
    });
});