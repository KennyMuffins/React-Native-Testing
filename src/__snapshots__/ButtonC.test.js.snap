// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonC Rendering should match to snapshot - Primary: Primary button snapshot 1`] = `ShallowWrapper {}`;

exports[`ButtonC Rendering should match to snapshot - Secondary: Secondary button snapshot 1`] = `ShallowWrapper {}`;
